### PR TITLE
Random spawn values

### PR DESCRIPTION
**Pull request by Evankhell561**
## About The Pull Request

This will generate random objects based on object variables instead of lists.

## Changelog
:cl:
add: Random spawn values
del: lists of random content.
/:cl:

### CODE REVIEW

File: code/game/objects/random/circuitboards.dm [hunk 1]
```diff
@@ -1,63 +1,15 @@
-/obj/random/circuitboard
+/obj/item/weapon/electronics
+	spawn_tags = SPAWN_TAG_ELECTRONICS
+	rarity_value = 20
+	spawn_frequency = 10
+	bad_types = /obj/item/weapon/electronics
```
Comment (TorinoFermic) [inline code/game/objects/random/circuitboards.dm:5]: We would like if you can find the original declaration to put these variables in /obj/item/weapon/electronics at proper file for circuit boards.

File: code/game/objects/random/packs.dm [hunk 2]
```diff
@@ -6,120 +6,94 @@ Packs are meant to be send mainly to junkpiles, but can be placed on map as well
 They generally give more random result and can provide more divercity in spawn.
 */
 
-/obj/random/pack/cloth
+/obj/spawner/pack/cloth
 	name = "Random cloth supply"
 	icon_state = "armor-red"
 	desc = "This is a random cloth supply."
+	tags_to_spawn = list(SPAWN_CLOTHING)
+	restristed_tags = list(SPAWN_VOID_SUIT)
```
Comment (TorinoFermic) [inline code/game/objects/random/packs.dm:14]: Misspelling : restricted

File: code/game/objects/random/traps/traps.dm [hunk 3]
```diff
@@ -1,35 +1,34 @@
...
-	/obj/item/weapon/beartrap/armed = 0.30,
-	/obj/item/weapon/beartrap/makeshift/armed = 0.45)
-
+	tags_to_spawn = list(SPAWN_TRAP_ARMED)
+
+/obj/spawner/traps/item_to_spawn()
+	.=..()
+	var/list/possible_traps = valid_candidates()
+	if(!islist(possible_traps))
+		admin_notice(SPAN_DANGER("NINGUNA TRAMPA HA SIDO ENCONTRADA"))//REMOVE IT
```
Comment (TorinoFermic) [inline code/game/objects/random/traps/traps.dm:11]: Forgot this line

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 4]
```diff
@@ -7,6 +7,8 @@
 	layer = BELOW_OBJ_LAYER
 	w_class = ITEM_SIZE_GARGANTUAN
 	matter = list(MATERIAL_STEEL = 10)
+	bad_types = "/obj/structure/closet"
```
Comment (TorinoFermic) [inline code/game/objects/structures/crates_lockers/closets.dm:10]: Why this line is enclosed in double quotes ?

Comment (Evankhell561) [reply code/game/objects/structures/crates_lockers/closets.dm:10]: for many reasons but none specific. The code also accepts it in this format.

Comment (TorinoFermic) [reply code/game/objects/structures/crates_lockers/closets.dm:10]: Okay

**Comment by GrayRachnid** [comment]
By the way does this actually impact anything ingame or is it all just code overhaul

**Comment by TorinoFermic** [comment]
> By the way does this actually impact anything ingame or is it all just code overhaul

It could impact the game by including everything that's wasnt blacklisted to the spawner, so anything we forgot to manually include will have in the spawner.

**Comment by TheShown911** [comment]
> By the way does this actually impact anything ingame or is it all just code overhaul

May lower resource consumption as well.

**Review by drexample — CHANGES_REQUESTED** [review]
@Evankhell561 Fix conflicts please

**Review by drexample — APPROVED** [review]

**Review by drexample — CHANGES_REQUESTED** [review]
EVEN MORE CONFLICTS GOD HELP

**Review by TorinoFermic — APPROVED** [review]
There's no way I could review all these changes

**Review by TheShown911 — APPROVED** [review]
This is gonna keep making conflicts untill the end of time, i will merge this the moment this conflict is resolved.

**Pull request merged**